### PR TITLE
Log errors and not using if the current database does not support transactions.

### PR DESCRIPTION
**Pull request by maliming**
From https://support.abp.io/QA/Questions/2176

eg: SQLite(`SqliteConnection does not support nested transactions`.).

### CODE REVIEW

File: framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/Uow/EntityFrameworkCore/UnitOfWorkDbContextProvider.cs [hunk 1]
```diff
@@ -170,19 +172,34 @@ private TDbContext CreateDbContextWithTransaction(IUnitOfWork unitOfWork)
...
+                    unitOfWork.AddTransactionApi(
+                        transactionApiKey,
+                        new EfCoreTransactionApi(
+                            dbtransaction,
+                            dbContext,
+                            _cancellationTokenProvider
+                        )
+                    );
+                }
+                catch (Exception e)
```
Comment (NecatiMeral) [inline framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/Uow/EntityFrameworkCore/UnitOfWorkDbContextProvider.cs]:
```csharp
catch (Exception e) when (e is InvalidOperationException || e is NotSupportedException)
{
    Logger.LogError(TransactionsNotSupportedErrorMessage);
    Logger.LogException(e);
}
```

Comment (NecatiMeral) [reply framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/Uow/EntityFrameworkCore/UnitOfWorkDbContextProvider.cs]: or maybe (to reduce code duplication)
```csharp
catch (Exception e) when (CatchAndLogTransactionNotSupportedException(e))
{
}

bool CatchAndLogTransactionNotSupportedException(Exception e)
{
  if (e is InvalidOperationException || e is NotSupportedException)
  {
      Logger.LogError(TransactionsNotSupportedErrorMessage);
      Logger.LogException(e);
  
      return true;
  }
  return false;
}
```

File: framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/Uow/EntityFrameworkCore/UnitOfWorkDbContextProvider.cs [hunk 2]
```diff
@@ -18,6 +18,8 @@ namespace Volo.Abp.Uow.EntityFrameworkCore
     public class UnitOfWorkDbContextProvider<TDbContext> : IDbContextProvider<TDbContext>
         where TDbContext : IEfCoreDbContext
     {
+        private const string TransactionsNotSupportedErrorMessage = "The current database does not support transactions, All operations will be performed in non-transactions, This may cause errors.";
```
Comment (hikalkan) [inline framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/Uow/EntityFrameworkCore/UnitOfWorkDbContextProvider.cs]:
```suggestion
        private const string TransactionsNotSupportedErrorMessage = "Current database does not support transactions. Your database may remain in an inconsistent state in an error case.";
```

File: framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/Uow/EntityFrameworkCore/UnitOfWorkDbContextProvider.cs (hunk 1 above)
Comment (hikalkan) [reply framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/Uow/EntityFrameworkCore/UnitOfWorkDbContextProvider.cs]: I agree with `when` usage.

**Review by hikalkan — APPROVED** [review]

**Pull request merged**